14 as a forgetful owner i can retrieve my account

jianyangg opened the pull request:
Close #14

jianyangg committed:
Add Recover Account Command

Close #14

jianyangg committed:
Fix checkstyle

jianyangg committed:
Add tests for recover account

jianyangg committed:
Add newline at EOF

B-enguin commented on src/main/java/seedu/address/logic/commands/user/UserRegisterCommand.java:
```diff
@@ -21,11 +23,14 @@ public class UserRegisterCommand extends Command {
             + ": Register an account for HomeBoss.\n"
             + "Parameters: "
-            + PREFIX_USER + " " + PREFIX_PASSWORD + " " + PREFIX_PASSWORD_CONFIRM + "\n"
+            + PREFIX_USER + " " + PREFIX_PASSWORD + " " + PREFIX_PASSWORD_CONFIRM
+            + " " + PREFIX_SECRET_QUESTION + " " + PREFIX_ANSWER + "\n"
```
Should we include the `PREFIX` with the value, e.g. `PREFIX_USER USER`

B-enguin commented on src/main/java/seedu/address/logic/parser/ParserUtil.java:
```diff
@@ -264,4 +264,28 @@ public static Password parsePassword(String password) throws ParseException {
     }
 
+    /**
+     * Parses a {@code String tag} into a {@code String}.
```
Should this be parses `{@code String secretQuestion}`

B-enguin commented on src/main/java/seedu/address/logic/parser/ParserUtil.java:
```diff
@@ -264,4 +264,28 @@ public static Password parsePassword(String password) throws ParseException {
+    }
+
+    /**
+     * Parses a {@code String tag} into a {@code String}.
```
Same as above

B-enguin commented on src/main/java/seedu/address/logic/parser/user/UserRecoverAccountCommandParser.java:
```diff
@@ -0,0 +1,71 @@
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public UserRecoverAccountCommand parse(String args) throws ParseException {
+        System.out.println(args);
```
Should this still be here, or replaced by logger

B-enguin commented on src/main/java/seedu/address/logic/parser/user/UserRecoverAccountCommandParser.java:
```diff
@@ -0,0 +1,71 @@
+        Password password = ParserUtil.parsePassword(argMultimap.getValue(PREFIX_PASSWORD).get());
+        Password confirmPassword = ParserUtil.parsePassword(argMultimap.getValue(PREFIX_PASSWORD_CONFIRM).get());
+        String answer = ParserUtil.parseAnswer(argMultimap.getValue(PREFIX_ANSWER).get());
+        if (!password.equals(confirmPassword)) {
+            throw new ParseException(UserRecoverAccountCommand.MESSAGE_PASSWORD_MISMATCH);
```
Could consider testing this path

B-enguin commented on src/main/java/seedu/address/model/ModelManager.java:
```diff
@@ -280,6 +280,18 @@ public void registerUser(User user) {
+    public void resetPassword(User user) {
+        // overrides the stored user
+        userPrefs.registerUser(user);
+        setLoggedInUser(user);
+        this.setLoggedInUser(user);
```
Are these duplicates

B-enguin commented on src/main/java/seedu/address/model/UserPrefs.java:
```diff
@@ -124,16 +124,18 @@ public User getStoredUser() {
-            // if authenticationData.getUsername() is null, then the file is empty
-            if (authenticationData.getUsername() == null) {
+            // if either field in the authentication file is empty, return null
+            if ((!Username.isValidUsername(authenticationData.getUsername()))) {
```
Which is the second field?

B-enguin reviewed (commented):
Some general comments

jianyangg committed:
Resolve comments in PR

Updated parameters of register and
recover account commands and removed
some debug print statements and
added more tests.

B-enguin reviewed (approved):
LGTM

jianyangg committed:
Merge branch 'master' into 14-as-a-forgetful-owner-i-can-retrieve-my-account

jianyangg committed:
Fix checkstyle

Removed whitespace
